### ISSUE TITLE
Prevent same songs from appearing in radio queue

### ISSUE THREAD

**Issue by aneesh1122** [fast4x/RiMusic#2102]
### Description

Steps
1) play radio of any song
2) play 48th or 49th song
3) new songs will be added in queue

https://github.com/fast4x/RiMusic/assets/45353488/2323e6cc-78bb-4dc2-8d80-e3e15aeb1e9d

### Suggested Solution

In these new added songs in 3), most of the songs are repeated from the first 49 songs.

https://github.com/fast4x/RiMusic/assets/45353488/aeb2bc07-a0c2-465a-b659-711972894fff

There are two ways to solve this
1) take 49 songs suggestions but those next 49 songs shouldn't include any songs from queue
2) take 49 songs suggestions but add n number of songs(<49) in the queue after removing all the songs which are already in queue

Basically there shouldn't be any repeated songs in queue

### Additional Context

If possible then please add an option to stop liked songs from appearing in radio suggestions. I use radio to discover new songs afterall

**Comment by PaolaAtilano** [comment]
I have the same problem too! I wonder if it's a problem with any API or something?

**Comment by aneesh1122** [comment]
> I have the same problem too! I wonder if it's a problem with any API or something?

YT API doesn't repeat songs in the radio queue.

It might have to do something with the unofficial API RiMusic uses.

**Comment by aneesh1122** [comment]
@fast4x could you please look into this once.

I'm sure it will help those people who use radio function

**Comment by aneesh1122** [comment]
Could you please look into this once? @fast4x 

Most of the radio suggestions are the songs I've already liked and most of the new suggestions are songs which are already in queue

Please add an option to prevent already liked songs from appearing in the radio suggestions and prevent duplicate songs in queue.
